midterm-2023-11-07/code/5042-ivan-apostolovski/script.ts

```diff
@@ -0,0 +1,358 @@
+    idCell.innerHTML = movie.id.toString();
+    row.appendChild(idCell);
+
+
+
+    // NAME CELL //
+    const nameCell = document.createElement("div");
+    nameCell.classList.add("movie-data", "movie-name");
+    nameCell.innerHTML = movie.title;
+    row.appendChild(nameCell);
+
+
+    // DIRECTOR CELL //
+    const directorCell = document.createElement("div");
+    directorCell.classList.add("movie-data", "movie-director");
+    directorCell.innerHTML = movie.director;
+    row.appendChild(directorCell);
+
+
+    //YEAR CELL////
+    const relaseCell = document.createElement("div");
+    relaseCell.classList.add("movie-data", "movie-relese");
+    relaseCell.innerHTML = movie.year.toString();
+    row.appendChild(relaseCell);
+
+
+    /////GENRE CELL///
+    const genreCell = document.createElement("div");
+    genreCell.classList.add("movie-data", "movie-genre");
+    genreCell.innerHTML = movie.genre.join(" / ");
+    row.appendChild(genreCell);
+
+
+    //////  PLOT CELL /////
+    const plotCell = document.createElement("div");
+    plotCell.classList.add("movie-data", "movie-plot");
+    const maxChar = 50;
+    let plotText = movie.plot;
+
+    if (plotText.length > maxChar) {
+        const lastword = plotText.lastIndexOf(" ", maxChar);
+        plotText = plotText.substring(0, lastword) + "...";
+    }
+    plotCell.innerHTML = plotText;
+
+    plotCell.addEventListener("click", () => {
+        const modal = document.getElementById("plot-details")!;
+        modal.classList.remove("hidden");
+
+        const modalHeader = document.querySelector("#plot-details-content h2")! as HTMLHeadingElement;
+        modalHeader.innerText = `Plot for: ${movie.title}`;
+
+        const plotParagraph = document.getElementById("movie-details-plot")!;
+        plotParagraph.innerText = movie.plot;
+    });
+    
+    row.appendChild(plotCell);
+
+
+    
+    /////CAST CELL ////
+    const actorsCell = document.createElement("div");
+    actorsCell.classList.add("movie-data", "movie-actors");
+    const actorList = document.createElement("ul");
+    actorList.id = "movie-details-actor-list"
+
+    for (const actor of movie.cast) {
+        const actorItem = document.createElement("li");
+        actorItem.innerText = `${actor.actor} (${actor.character})`;
+        genreCell.innerHTML = movie.genre.join(" / ");
+        actorList.appendChild(actorItem);
+    }
+    actorsCell.appendChild(actorList);
+    row.appendChild(actorsCell);
+
+
+
+    //Oscars CELL //
+    const oscarsCell = document.createElement("div");
+    oscarsCell.classList.add("movie-data", "movie-oscars");
+
+
+    if (movie.oscars) {
+        const numOscars = Object.values(movie.oscars).length
+        oscarsCell.innerHTML = numOscars.toString();
+    }
+    else {
+        oscarsCell.innerHTML = "0"
+    }
+
+
+    row.appendChild(oscarsCell);
+
+    return row;
+}
+
+function refreshPage(){
+    window.location.reload();
+} 
+
+if (document) {
```

Comment:
This can also be done in the `DOMContentLoaded` handler - this way is functional, but a bit outdated.